cabin-sorting/api/sortingData.js

```diff
@@ -0,0 +1,93 @@
+import path from 'path'
+import fs from 'fs'
+import { client, connectToDatabase } from './mongodb'
+
+let validatedHackers
+
+// Get all valid hacker data from the applicant_data database and convert to json contents
+async function grabFromDatabase () {
+  const database = connectToDatabase()
+  const applicantData = await database.collection('applicant_data')
+
+  try {
+    // Query for hackers whose application status is submitted and who has a
+    // 'Post Acceptance Response' section
+    const query = {
+      applicationStatus: 'Submitted',
+      postAcceptanceResponses: { $exists: true },
+      isAdmin: false,
+      rsvpStatus: 'Confirmed',
+      'postAcceptanceResponses.swag': { $exists: false },
+      'postAcceptanceResponses.club': { $exists: true }
+    }
+
+    // Find all valid hackers that have answers to the cabin questions
+    const hackerDataCursor = await applicantData
+      .find(query)
+      .project({ email: 1, postAcceptanceResponses: 1 })
+    const validHackers = []
+    // create list of hackers with their information/questions
+    while (await hackerDataCursor.hasNext()) {
+      let item = await hackerDataCursor.next()
+
```

Comment:
A map instead of a while loop might be cleaner

Reply:
You can utilize `Promise.all`! Here some docs https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Reply:
I realized you can do the Cursor.toArray() which can let us avoid the whole map/next() situation